### ISSUE TITLE
Add a 'wrap' mode to Animation and AnimationWidget

### ISSUE THREAD

**Issue by alisterburt** [sofroniewn/napari-animation#13]
What do we think about a 'wrap' mode for animations, you define the 'forward' pass and the rendered animation also contains the reverse for a nice loop.

I was thinking a bool attribute of the `Animation` class, exposed by a checkbox in the `AnimationWidget`.

That bool being set to True would modify a copy of the keyframe dict within `Animation.animate` to provide the reverse direction.

This approach is naive and would involve rerendering every frame but I don't know how I would reuse frames with the current state generator

**Comment by sofroniewn** [comment]
Ah to make "continuous movies"? I'd probably think about an enum. You might also just want a "loop" where you add the first frame on to the end of the list so that it ends and starts in the same place without doing the forward and reverse, though I can see how both are desirable.

For going in "reverse" I'd try to avoid copying the whole keyframe dict, but maybe you can just traverse it backwards with appropriate flags/ settings inside `_state_generator` which could now take a `reverse=False` bool input and if true you could do a `1-fraction` in the appropriate place. Inside `_frame_generator` you could then call `_state_generator` twice, or add in that extra frame from the beginning depending on the "mode"

**Comment by alisterburt** [comment]
Yeah exactly - that sounds good! I'll play with this once I'm a bit further
on the `KeyFramesListWidget`
